commit 2fdc1dbb5fbca529b44f030b6811517d86425d6c (refs/changes/26/2326/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-08-04T00:20:15+00:00 (2 years, 2 months ago)
    
    joyent/node-cueball#118 slot should retain reference to previous claim handle
    Reviewed by: David Pacheco <david.pacheco@joyent.com>
    Reviewed by: Jordan Hendricks <jordan.hendricks@joyent.com>

diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index 4588ebf..602596c 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -872,6 +872,7 @@ function ConnectionSlotFSM(options) {
 	this.csf_backend = options.backend;
 	this.csf_wanted = true;
 	this.csf_handle = undefined;
+	this.csf_prevHandle = undefined;
 	this.csf_monitor = options.monitor;
 
 	this.csf_checker = options.checker;
@@ -1008,6 +1009,8 @@ ConnectionSlotFSM.prototype.state_idle = function (S) {
 	var self = this;
 	var smgr = this.csf_smgr;
 
+	if (this.csf_handle !== undefined)
+		this.csf_prevHandle = this.csf_handle;
 	this.csf_handle = undefined;
 
 	/* Monitor successfully connected: make it into a normal slot now. */
